### PR TITLE
Testcases for RSBackup.dll

### PR DESCRIPTION
**Pull request by VinothJeyaraman**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as a link to the swagger spec, used to generate the code.
- [ ] The `project.json` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by shahabhijeet** [comment]
@VinothJeyaraman add to whitelist

**Comment by shahabhijeet** [comment]
@VinothJeyaraman please use the link above and get CLA approved.

**Comment by VinothJeyaraman** [comment]
@shahabhijeet , I am a Microsoft FTE. I have just linked my account to Microsoft. 
Is CLA still required?

**Comment by VinothJeyaraman** [comment]
Hi,

I am a Microsoft FTE and I have given details in CLA form. I am yet to get the mail from DocuSign to sign electronically. 
Can someone tell me what should I do further?

File: src/SDKs/RecoveryServices.Backup/RecoveryServices.Backup.Tests/ScenarioTests/DpmTests.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    ///	                    - DpmProductionServerName format is ContainerType;ContainerName
+    ///             "DpmBackupEngineName": "VinoVenus.DPMDOM02.SELFHOST.CORP.MICROSOFT.COM"
+    ///                     - DpmBackupEngineName format is BackupEngineName
+    /// </summary>
+    public class DpmTests : TestBase, IDisposable
+    {
+        private const string dpmVaultName = "SDKTestDPMVault";
+        private const string dpmResourceGroupName = "SDKTestDPMRg";
+
+        [Fact]
```
Comment (shahabhijeet) [inline src/SDKs/RecoveryServices.Backup/RecoveryServices.Backup.Tests/ScenarioTests/DpmTests.cs]: @VinothJeyaraman please record all new tests using latest version of Resource Manager or latest version of dependent ARM sdk packages.
We had a lots of issue migrating majority of tests when we moved to VS2017, so we are asking everyone who are recording tests to record using latest SDK packages

File: src/SDKs/RecoveryServices.Backup/RecoveryServices.Backup.Tests/TestHelpers/TestSettings.cs [hunk 2]
```diff
@@ -33,6 +46,13 @@ public TestSettings Initialize()
             VirtualMachineType = testSettings.VirtualMachineType;
             RestoreStorageAccountName = testSettings.RestoreStorageAccountName;
             RestoreStorageAccountResourceGroupName = testSettings.RestoreStorageAccountResourceGroupName;
+			AzureSqlContainerName = testSettings.AzureSqlContainerName;
```
Comment (shahabhijeet) [inline src/SDKs/RecoveryServices.Backup/RecoveryServices.Backup.Tests/TestHelpers/TestSettings.cs]: nit: fix tabs

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Please update the version number for your SDK package.
Please remove any traces of project.json, global.json and *.xproj, these files are no longer needed for the new project system.
Also add a description for your change.

**Comment by shahabhijeet** [comment]
@VinothJeyaraman please let me know if you are not clear as to what needs to be done.
In short, by default your tests are using 1.1.0-preview version of nuget package.
I am asking you to start recording new tests using 1.6.0 version of Resource Manager nuget package.
This will allow you to move away from a basic nuget package that is really very old and we should not be using it anyway to begin with at this stage.
So you will want to add a reference to the new version in your test project and start using the new Resource Manager client Side by Side in your tests. Eventually you would want to re-record all your tests with the latest resource manager client.

**Comment by VinothJeyaraman** [comment]
@shahabhijeet 
Hi Abhijeet, Could you please tell me how to update nugget package? I tried with update-package in package Manager console. It says 'update not available'.

**Comment by markcowl** [comment]
@VinothJeyaraman Y&ou can simply update the version in the csproj.  1.6.0-preview is the latest, I believe.

**Comment by VinothJeyaraman** [comment]
@shahabhijeet , @markcowl 
I don't find the package reference in csproj file. 

However, I found that in AzSdk.test.reference.props. When I changed the version number and tried to build it, I got the following error :  Unable to resolve 'Microsoft.Azure.ResourceManager (>= 1.6.0)' for '.NETCoreApp,Version=v1.1'.	

can you please help me out?

**Comment by shahabhijeet** [comment]
@VinothJeyaraman add a reference to the nuget package to your test project and then load your project that would add the new reference.
E.g. Add this to your test project and reload your csproj file in VS
<ItemGroup> 
	<PackageReference Include="Microsoft.Azure.Management.ResourceManager" Version="1.6.0-preview" /> 
</ItemGroup> 
You do not need to change it in AzSdk.test.reference.props, that is for backward compatibility.
Once you do this, the new nuget package has new namespace and that would help you start creating new ResourceManagmentClient using the new nuget package.

**Comment by shahabhijeet** [comment]
@VinothJeyaraman please reopen PR once you are ready to merge

**Pull request closed without merging**